source/opt/add_var_for_funccall_param.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/opt/add_var_for_funccall_param.h"
+
+#include "ir_builder.h"
+
+using namespace spvtools;
+using namespace opt;
+
+Pass::Status AddVarsForFuncCallParamPass::Process() {
+  auto result = Status::SuccessWithoutChange;
+  auto funcsNum = get_module()->end() - get_module()->begin();
+  if (funcsNum == 1) return result;
```

Comment:
This code is hard to understand .. first of all one method contains so many lines .. each line of code is the low-level detail without description.

For example, I guess you checked `if (funcsNum == 1) return result;` to know that it does not have any other function to call. Am I right?

If the code was
```
// Returns true if there is only single function in the module so no function call exists.
bool AddVarsForFuncCallParamPass::HasNoFunctionToCall() { .. }
```
It would be clear to understand your intention.